src/Core/BaseSettings.h

```diff
@@ -43,9 +43,14 @@ class BaseSettings : public TTraits::Data
 {
     using CustomSettingMap = std::unordered_map<std::string_view, std::pair<std::shared_ptr<const String>, SettingFieldCustom>>;
 public:
+    BaseSettings() = default;
+    BaseSettings(const BaseSettings &) = default;
+    BaseSettings & operator=(const BaseSettings &) = default;
```

Comment:
what about move constructor/assignment?

Reply:
We don't use move constructor/assignment anywhere, so I think no need to add them. I added these constructors only because I added virtual destructor (because I made a virtual method) and I had to define these constructors/assignments too.

Reply:
https://en.cppreference.com/w/cpp/language/rule_of_three
I think it's still nice to follow rule of five, but it's fine as it is.